include/dfu/dfu_target.h

```diff
@@ -38,6 +38,8 @@ enum dfu_target_image_type {
 	DFU_TARGET_IMAGE_TYPE_SMP = 8,
 	/** SUIT Envelope */
 	DFU_TARGET_IMAGE_TYPE_SUIT = 16,
+	/** Custom update implementation, e.g. for external MCU */
+	DFU_TARGET_IMAGE_TYPE_CUSTOM = 32,
```

Comment:
This is already assigned to `DFU_TARGET_IMAGE_TYPE_SMP`, so comment would need to be updated to indicate it is **not** for external MCUs, and ID should be moved to be at the end of the list, if it's an 8-bit value then 128 etc. to allow for additional entries to be added here